app/utils/LocationFinder.scala

```diff
@@ -37,17 +36,30 @@ object LocationFinder {
   implicit val context = scala.concurrent.ExecutionContext.Implicits.global
 
 
-  // Find Euclidean distance between two GPS coordinates
+  // Find haversine distance between two GPS coordinates
   def dist(a: Location, b: Location): Double = {
-    sqrt(pow(a.lat - b.lat, 2) + pow(a.lng - b.lng, 2))
+    val a_lat_rad = Math.toRadians(a.lat)
+    val a_lng_rad = Math.toRadians(a.lng)
+
+    val b_lat_rad = Math.toRadians(b.lat)
+    val b_lng_rad = Math.toRadians(b.lng)
+
+    def haversine(theta: Double): Double = {
```

Comment:
could we use an anonymous function instead of def'ing one? I think it makes it more clear that this is a one-time function. idk, this is just my usual preferred way of doing this sort of thing in python
`val haversine = (theta: Double: => Math.pow(Math.sin(theta/2), 2)`